### DISCUSSION TITLE
Claude dev just messed up my git history for a very large piece of work

### DISCUSSION THREAD

**Discussion by MRiabov** [cline/cline#317] [Feature Requests]
I've had a file for 2000 lines of code. I asked Claude to refactor something, and it output something with 136, that didn't even fix the issue... I accidentally pressed "enter" and I guess it accepted the changes.

Now, the git can't show the history of my edits, and I'm stuck with 136 line file.
Git commit history also refuses to show up.
![image](https://github.com/user-attachments/assets/afe12858-c3f3-4e19-b911-65475a2e7e73)
Make your software safer, please.

**Comment by quinn** [comment]
This seems unlikely to be related to Claude Dev. Can you show the chat history in the extension? What happens when you run `git status`?

**Comment by saoudrizwan** [comment]
I've created a wiki to help with when claude deletes code like this: https://github.com/saoudrizwan/claude-dev/wiki/Troubleshooting-%E2%80%90-Claude-Deleting-Code-with-%22rest-of-code-here%22-Comments

**Reply by MRiabov** [reply]
It would be useful if Claude dev automatically preserved the files which you send to Claude. Similarly to how it works on claude's or chatGPT's UI - you send a message and it permanently saves it in the storage.
This would prevent the losses.
